src/analyzer/protocol/ssl/SSL.cc

```diff
@@ -98,4 +174,166 @@ void SSL_Analyzer::Undelivered(uint64_t seq, int len, bool orig)
 	interp->NewGap(orig, len);
 	}
 
+void SSL_Analyzer::SetSecret(const u_char* data, int len)
+	{
+	// FIXME: Is this the proper way to initialize a zeek::StringVal?
+	//        the alternative requires a cast: new zeek::StringVal(len, data)
+	secret = new zeek::StringVal(new zeek::String(data, len, true));
```

Comment:
Optimally, it would be neat to make this an IntrusivePtr - then you don't have to worry about cleaning it up - or this function being called several times.

Apart from that - personally I would prefer the cast from u_char to char, without goint through zeek::String.